test/integration_test.go

```diff
@@ -276,6 +276,56 @@ func TestClientAPI(t *testing.T) {
 	assert.Contains(string(secret), `{"status":"success","data":{"symmetric_key_shared":{"Type":"symmetric-key","Size":128,`)
 }
 
+func TestSettingSecrets(t *testing.T) {
+	assert := assert.New(t)
+	require := require.New(t)
+
+	// start Coordinator
+	cfg := newCoordinatorConfig()
+	defer cfg.cleanup()
+	coordinatorProc := startCoordinator(cfg)
+	require.NotNil(coordinatorProc, "could not start coordinator")
+	defer coordinatorProc.Kill()
+
+	log.Println("Setting the Manifest")
+	_, err := setManifest(testManifest)
+	require.NoError(err, "failed to set Manifest")
+
+	// create client with certificates
+	privk, err := x509.MarshalPKCS8PrivateKey(RecoveryPrivateKey)
+	require.NoError(err)
+	clCert, err := tls.X509KeyPair(AdminCert, pem.EncodeToMemory(&pem.Block{Type: "PRIVATE KEY", Bytes: privk}))
+	require.NoError(err)
+	client := http.Client{Transport: &http.Transport{TLSClientConfig: &tls.Config{
+		Certificates:       []tls.Certificate{clCert},
+		InsecureSkipVerify: true,
+	}}}
+
+	// start server
+	log.Println("Starting a Server-Marble...")
+	serverCfg := newMarbleConfig(meshServerAddr, "test_marble_server", "server,backend,localhost")
+	defer serverCfg.cleanup()
+	serverProc := startMarbleServer(serverCfg)
+	require.NotNil(serverProc, "failed to start server-marble")
+	defer serverProc.Kill()
+
+	// start a marble
+	log.Println("Starting a Client-Marble with unset secret, this should fail...")
+	clientCfg := newMarbleConfig(meshServerAddr, "test_marble_unset", "client,frontend,localhost")
+	defer clientCfg.cleanup()
+	assert.False(startMarbleClient(clientCfg))
+
+	// test setting a secret
+	log.Println("Setting a custom secret")
+	clientAPIURL := url.URL{Scheme: "https", Host: clientServerAddr, Path: "secrets"}
+	_, err = client.Post(clientAPIURL.String(), "application/json", bytes.NewReader([]byte(UserSecrets)))
```

Comment:
what's the difference between this and `strings.NewReader(UserSecrets)`?

Reply:
No real difference, i saw `bytes.NewReader` being used everywhere else in the integration test and switched it.

Reply:
AFAIK it's best to use the one that doesn't require a cast

Reply:
Makes sense, I'll look over the code and see where the `strings` function can be used over `bytes`